dev_tools/nyaml2nxdl/nyaml2nxdl_forward_tools.py

```diff
@@ -0,0 +1,1242 @@
+            if "__line__" in attr:
+                continue
+            line_number = f"__line__{attr}"
+            line_loc = value[line_number]
+            if attr in ["doc", *attr_attr_list] and not isinstance(attr_val, dict):
+                if attr == "unit":
+                    elemt_obj.set(f"{attr}s", str(value[attr]))
+                    rm_key_list.append(attr)
+                    rm_key_list.append(line_number)
+                    xml_handle_comment(obj, line_number, line_loc, elemt_obj)
+                elif attr == "exists" and attr_val:
+                    xml_handle_exists(value, elemt_obj, attr, attr_val)
+                    rm_key_list.append(attr)
+                    rm_key_list.append(line_number)
+                    xml_handle_comment(obj, line_number, line_loc, elemt_obj)
+                elif attr == "doc":
+                    xml_handle_doc(
+                        elemt_obj, format_nxdl_doc(attr_val), line_number, line_loc
+                    )
+                    rm_key_list.append(attr)
+                    rm_key_list.append(line_number)
+                else:
+                    elemt_obj.set(attr, check_for_mapping_char_other(attr_val))
+                    rm_key_list.append(attr)
+                    rm_key_list.append(line_number)
+                    xml_handle_comment(obj, line_number, line_loc, elemt_obj)
+
+        for key in rm_key_list:
+            del value[key]
+        # Check cor skiped attribute
+        check_for_skiped_attributes("Attribute", value, attr_attr_list, verbose)
+    if value:
+        recursive_build(elemt_obj, value, verbose)
+
+
+def validate_field_attribute_and_value(v_attr, vval, allowed_attribute, value):
+    """
+    Check for any attributes that comes with invalid name,
+        and invalid value.
+    """
+
+    # check for empty val
+    if not isinstance(vval, dict) and not str(vval):  # check for empty value
+        line_number = f"__line__{v_attr}"
+        raise ValueError(
+            f"In a field a valid attrbute ('{v_attr}') found that is not stored."
+            f" Please check arround line {value[line_number]}"
+        )
+
+    # The bellow elements might come as child element
+    skipped_child_name = ["doc", "dimension", "enumeration", "choice", "exists"]
+    # check for invalid key or attributes
+    if (
+        v_attr not in [*skipped_child_name, *allowed_attribute]
+        and "__line__" not in v_attr
+        and not isinstance(vval, dict)
+        and "(" not in v_attr  # skip only groups and field that has name and type
+        and "\\@" not in v_attr
+    ):  # skip nexus attributes
+        line_number = f"__line__{v_attr}"
+        raise ValueError(
+            f"In a field or group a invalid attribute ('{v_attr}') or child has found."
+            f" Please check arround line {value[line_number]}."
+        )
+
+
+def xml_handle_fields(obj, keyword, value, line_annot, line_loc, verbose=False):
```

Comment:
Code is very similar to `xml_handle_group` above

Reply:
Yes, it is intended. I tried to have the same structure of the function body, for better maintainability. So, any changes and/or updates in any concept like, group, field, attributes, can be implemented easily and in a independent manner.

Reply:
I disagree: if they are largely the same then to be maintainable, you should refactor to reduce the code!